pkg/credentials/credentials.go

```diff
@@ -265,22 +326,3 @@ func defaultValidationFunc(creds map[string]string) error {
 	}
 	return nil
 }
-
-func openstackValidationFunc(creds map[string]string) error {
```

Comment:
I would keep this validation function because the validation in the library is not as strict as our validation. A couple of examples:

* the validation would pass if [`OS_USERID` is set instead of `OS_USERNAME`](https://github.com/gophercloud/gophercloud/blob/c8c96f74de7f0c91297732d1a1468bc225a68137/openstack/auth_env.go#L67-L75), but we don't support `OS_USERID`
* the validation would pass if [`OS_PASSCODE` is set instead of `OS_PASSWORD`](https://github.com/gophercloud/gophercloud/blob/c8c96f74de7f0c91297732d1a1468bc225a68137/openstack/auth_env.go#L77-L83)